src/interpreter.h

```diff
@@ -227,25 +227,42 @@ struct InterpreterHostImportDelegate {
 };
 
 struct InterpreterModule {
+  explicit InterpreterModule(bool is_host);
+  InterpreterModule(const StringSlice& name, bool is_host);
+  virtual ~InterpreterModule();
+
+  inline struct DefinedInterpreterModule* as_defined();
+  inline struct HostInterpreterModule* as_host();
+
   StringSlice name;
   InterpreterExportVector exports;
   BindingHash export_bindings;
   uint32_t memory_index; /* INVALID_INDEX if not defined */
   uint32_t table_index;  /* INVALID_INDEX if not defined */
   bool is_host;
-  union {
-    struct {
-      InterpreterImportArray imports;
-      uint32_t start_func_index; /* INVALID_INDEX if not defined */
-      size_t istream_start;
-      size_t istream_end;
-    } defined;
-    struct {
-      InterpreterHostImportDelegate import_delegate;
-    } host;
-  };
 };
-WABT_DEFINE_VECTOR(interpreter_module, InterpreterModule);
+struct DefinedInterpreterModule : InterpreterModule {
```

Comment:
For now it feels like it will be visual noise, since all the user code expects the members to be public anyway.